src/main/java/Duke.java

```diff
@@ -1,10 +1,128 @@
+            event.addToTaskList(errand, timestamp);
+            break;
+        case "DEADLINE":
+            errand = getErrand(userInput);
+            timestamp = getTimestamp(userInput);
+            timestampHeader = getTimestampHeader(userInput);
+            Task deadline = new Deadline(errand, timestamp, timestampHeader);
+            deadline.addToTaskList(errand, timestamp);
+            break;
+        case "LIST":
+            Task.printList();
+            break;
+        case "DONE":
+            errand = getSubstring(userInput);
+            Task.markDone(errand); // In this case, errand is the index of the item
+            break;
+        case "BYE":
+            printGoodbyeMessage();
+            break;
+        }
+    }
+
+    private static String getSubstring(String userInput) {
+        int spacePosition = userInput.indexOf(" ");
+        String inputSubstring = userInput.substring(spacePosition+1);
+        return inputSubstring;
+    }
+
+    private static String getErrand(String userInput) {
+        String inputSubstring = getSubstring(userInput);
+        int slashPosition = inputSubstring.indexOf("/");
+        String errand = inputSubstring.substring(0, slashPosition);
+        return errand.trim();
+    }
+
+    private static String getTimestamp(String userInput) {
+        String inputSubstring = getSubstring(userInput);
+        int slashPosition = inputSubstring.indexOf("/");
+        String timestamp = inputSubstring.substring(slashPosition);
+        int spacePosition = timestamp.indexOf(" ");
+        timestamp = timestamp.substring(spacePosition);
+        return timestamp.trim();
+    }
+
+    private static String getTimestampHeader(String userInput){
+        String inputSubstring = getSubstring(userInput);
+        int slashPosition = inputSubstring.indexOf("/");
+        String timestampHeader = inputSubstring.substring(slashPosition);
+        int spacePosition = timestampHeader.indexOf(" ");
+        timestampHeader = timestampHeader.substring(1, spacePosition);
+        return timestampHeader;
+    }
+
+    /**
+     * Starts the Task Manager program.
+     * <p>Runs an infinite loop until "BYE" is called</p>
+     */
+    private static void runTaskManager() {
+        String userInput;
+        String command;
+        Scanner in = new Scanner(System.in);
+
+        do {
+            userInput = in.nextLine();
+            command = getCommand(userInput);
+            executeCommand(command, userInput);
+        } while (!command.equals("BYE"));
+    }
+
     public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+        initialiseWelcomeMessage();
+        runTaskManager();
```

Comment:
I like how you extract the methods out for the main program. I see this in other places as well.